gpflow/base.py

```diff
@@ -117,14 +116,27 @@ def __init__(
         therefore we need a positive constraint and it is natural to use constrained values.
         A prior can be imposed either on the constrained version (default) or on the unconstrained version of the parameter.
         """
-        if transform is None:
-            transform = tfp.bijectors.Identity()
+        if isinstance(value, Parameter):
+            transform = transform if transform else value.transform
+            prior = prior if prior else value.prior
+            prior_on = prior_on if prior_on else value.prior_on
+            trainable = trainable if trainable is not None else value.trainable
+            name = name if name else value.bijector.name
+
+            if dtype:
+                value = _cast_to_dtype(value, dtype)
+        else:
+            if transform is None:
+                transform = tfp.bijectors.Identity()
+
+            prior_on = prior_on if prior_on else PriorOn.CONSTRAINED
+            trainable = trainable if trainable is not None else True
 
-        value = _cast_to_dtype(value, dtype)
-        _validate_unconstrained_value(value, transform, dtype)
+            value = _cast_to_dtype(value, dtype)
+            _validate_unconstrained_value(value, transform, dtype)
```

Comment:
Is there a risk that we're now not always validating the parameter values?  (though there are times when I don't care about invalid values.)

Reply:
Good point. If the transformation changed then the value (which was valid under the previous transformation) might no longer be valid. I will move this check so it is always called.